lib/proc.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from subprocess import CalledProcessError
+from subprocess import run as subprocess_run
+from typing import Tuple
+from typing import Union
+
+
+def run(
+    cmd: Tuple[str, ...], exit: bool = False, **kwargs: Union[str, bool, dict[str, str]]
+) -> str:
+    kwargs["check"] = True
+    kwargs["capture_output"] = True
+    try:
```

Comment:
Could `run()` call `run_stream_output()` with the extra kwargs if `run_stream_output()` returns the result of its subprocess.run() call?

Would remove some of the duplication.

Reply:
i'm not sure how to deduplicate this in a nice way - if i could have i would have - run_stream_output would have to reraise the CalledProcessError if capture_output=True and run would still need to print more detail so overall not much is being deduplicated really